gateway/modules/auth/match.go

```diff
@@ -96,9 +108,14 @@ func (m *Module) matchQuery(ctx context.Context, project string, rule *config.Ru
 	if err != nil {
 		return nil, formatError(ctx, rule, err)
 	}
-	err = utils.StoreValue(ctx, "args.result", data, args)
-	if err != nil {
-		return nil, formatError(ctx, rule, err)
+	if rule.Store != "" {
+		if err := utils.StoreValue(ctx, rule.Store, data, args); err != nil {
+			return nil, formatError(ctx, rule, err)
+		}
+	} else {
+		if err := utils.StoreValue(ctx, "args.result", data, args); err != nil {
+			return nil, formatError(ctx, rule, err)
+		}
 	}
```

Comment:
You can improve on the logic. 
Set the value of  `rule.Store` to `args.result` when empty.